tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Lifecycle Environment')
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat, function_leapp_cv, module_leapp_lce, module_sca_manifest_org, upgrade_path
+):
+    logger.info(
+        'Creating Leapp Activation Key for RHEL %s -> %s',
+        upgrade_path['source_version'],
+        upgrade_path['target_version'],
+    )
+    # need to publish and promote cv on lce to get content available on client host
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
+    function_leapp_cv = function_leapp_cv.read()
+    ak = module_target_sat.api.ActivationKey(
+        content_view=function_leapp_cv,
+        environment=module_leapp_lce,
+        organization=module_sca_manifest_org,
+    ).create()
+    # In case of 'rhel7_server_extras' repos, need to enabled (overridden) repository
+    # to get content available on client host
+    if upgrade_path['source_version'].split('.')[0] == '7':
+        ActivationKey.content_override(
+            {
+                'id': ak.id,
+                'content-label': RHEL_REPOS['rhel7_server_extras']['id'],
+                'value': 'true',
+            }
+        )
+    return ak
+
+
+@pytest.fixture
+def verify_target_repo_on_satellite(
+    module_target_sat,
+    function_leapp_cv,
+    module_sca_manifest_org,
+    module_leapp_lce,
+    upgrade_path,
+    function_leapp_ak,
+):
+    """Verify target rhel version repositories has enabled on Satellite Server"""
+    target_rhel_major_ver = upgrade_path['target_version'].split('.')[0]
+    cmd_out = Repository.list(
+        {
+            'search': f'content_label ~ rhel-{target_rhel_major_ver}',
+            'content-view-id': function_leapp_cv.id,
+            'organization-id': module_sca_manifest_org.id,
+            'lifecycle-environment-id': module_leapp_lce.id,
+        }
+    )
+    repo_names = [out['name'] for out in cmd_out]
+    if target_rhel_major_ver == '9':
+        assert RHEL_REPOS['rhel9_bos']['name'] in repo_names
+        assert RHEL_REPOS['rhel9_aps']['name'] in repo_names
+    else:
+        assert RHEL_REPOS['rhel8_bos']['name'] in repo_names
+        assert RHEL_REPOS['rhel8_aps']['name'] in repo_names
+
+
+@pytest.fixture
+def register_host_with_satellite(
+    module_target_sat,
+    custom_leapp_host,
+    module_sca_manifest_org,
+):
+    """Register content host with satellite"""
+    logger.info(
+        'Registering Leapp Host with Satellite - RHEL %s', custom_leapp_host.os_version.major
+    )
+    result = custom_leapp_host.register(
+        module_sca_manifest_org, None, function_leapp_ak.name, module_target_sat
```

Comment:
Where is `function_leapp_ak` coming from? It is not defined in args.

Reply:
I need to check local changes

Reply:
That was just a merging issue, fixed now.

Reply:
resolved